src/spriteset_battle.cpp

```diff
@@ -31,11 +31,11 @@
 
 Spriteset_Battle::Spriteset_Battle() {
 	// Create background
-	if (!Game_Temp::battle_background.empty()) {
-		background.reset(new Background(Game_Temp::battle_background));
-	} else {
+	if (Game_Battle::GetTerrainId() > 0) {
 		// Background verifies that the Terrain ID is valid
 		background.reset(new Background(Game_Battle::GetTerrainId()));
+	} else {
+		background.reset(new Background(Game_Temp::battle_background));
```

Comment:
Did you verify this still works for event commands which start a battle with a fixed battle background?

Reply:
havn't verified anything yet as this is the not-fun part